src/rdiffbackup/locations/repository.py

```diff
@@ -267,6 +286,79 @@ def _is_writable(self):
                 return False
         return True
 
+    def _is_locked(self, remove=False):
+        """
+        Validate if the repository is locked or not by the file
+        'rdiff-backup-data/.lock'
+
+        Returns True if the file exists, else returns False
+        """
+        # we need to make sure we have the last state of the lock
+        self.lock.setdata()
+        if self.lock.lstat():
+            if self.force:
+                log.Log("Repository is locked but forcing action nevertheless",
+                        log.WARNING)
+                if remove:
+                    self.lock.delete()
+                return False
+            else:
+                log.Log("Repository is locked by file {lf}, another "
+                        "action is probably on-going. Either wait, remove "
+                        "the lock or use the --force option".format(
+                            lf=self.lock), log.ERROR)
+                return True
+
+    def _lock(self):
+        """
+        Write a specific file 'rdiff-backup-data/.lock' to grab the lock,
+        and verify that no other process took the lock by comparing its
+        content.
+
+        Return True if the lock could be taken, False else.
+        """
+        fqdn = socket.getfqdn()
+        currtime = Time.curtimestr
+        pid = os.getpid()
+        cmd = sys.argv[0]
+        identifier = "{} {} {} {}".format(fqdn, currtime, pid, cmd)
```

Comment:
If you are doing this.

Somewhere in the code we write the pid to current_mirror. That become useless don't you think ?

Reply:
Yeah. I didn't change it for backward compatibility reasons.